GliaWidgets/Sources/EntryWidget/EntryWidget.swift

```diff
@@ -172,7 +172,50 @@ extension EntryWidget: UIViewControllerTransitioningDelegate {
             presentedViewController: presented,
             presenting: presenting,
             height: height,
-            cornerRadius: theme.entryWidget.cornerRadius
+            cornerRadius: environment.theme.entryWidget.cornerRadius
         )
     }
 }
+
+private extension EntryWidget {
+    func handleQueuesMonitorUpdates(state: QueuesMonitor.State) {
+        switch state {
+        case .idle:
+            break
+        case .updated(let queues):
+            let availableChannels = resolveAvailableChannels(from: queues)
+            self.channels = availableChannels
+        case .failed(let error):
+            // TODO: Handle error on EntryWidgetView
+            print(error)
+        }
+    }
+
+    func resolveAvailableChannels(from queues: [Queue]) -> [Channel] {
```

Comment:
As I understand, usage of `Set` makes sure that channels are unique.
Is it necessary to return `Array` here, instead of a `Set` used inside? If conversation to `Array` needed for consumer of the `channels`, it probably makes sense to convert at consumer side, otherwise it is not clear why we convert it here to `Array`.

Reply:
On UI it will be needed to sort the Channels' list, so that's why it's returning `Array`.

Reply:
Does it make sense to return sorted array to preserve some kind of oder?

Reply:
I don't know which order is required on the UI side, I will left a TODO comment so @rasmustautsglia will sort it in a way needed